### DISCUSSION TITLE
Print from CMD line

### DISCUSSION THREAD

**Discussion by joaodaffonsojr** [sumatrapdfreader/sumatrapdf#4492] [Q&A]
I have an application that needs to print PDFs via the command line. I'm using Sumatra for this job and it's very fast for this. However, a 310 kb PDF after being printed by Sumatra ends up with a very large size in the print spool, weighing 118 Mb. Is there a command to reduce this size?

This is the cmd line that I´m using
"C:\Users\Joao D Affonso Jr\AppData\Local\SumatraPDF\SumatraPDF.exe"  -print-to "EPSON L3150 Series PB" "D:\teste.pdf"

![Screenshot 2024-08-22 074735](https://github.com/user-attachments/assets/32628a89-b2d4-4f6a-9d22-8ea6b34e5aba)
![Screenshot 2024-08-22 074754](https://github.com/user-attachments/assets/0595a05b-f562-4ebc-a955-0534d15c9754)

**Comment by GitHubRulesOK** [comment]
Not really. It is down to the driver so each printer driver will produce different sizes 
Epson L3150 is exceptionally high resolution for an inkjet (5760 x 1440 DPI) thus the images sent will naturally be larger than another device.

SumatraPDF does not control the output it hands the pages over to windows printing for print as images .

Printer drivers often have some means to reduce quality thus if each dimension is halved the pages will then only use 1/4 the memory.

As an example take Windows print drivers test page thus a constant source for that driver and I change the print drivers settings the size of the spool file will alter. It is not the application (Windows test file) nor the printer device but the driver that alters resoloution. This is my inkjet.
![image](https://github.com/user-attachments/assets/628a0f5c-5844-4a6e-907b-eb98e3afc3bc)

Same windows test print in a PDF virtual printer
![image](https://github.com/user-attachments/assets/b51ee3e1-889b-408d-873f-4c54f2e1e693)

**Comment by GitHubRulesOK** [comment]
There was only one way SumatraPDF had influence over output and that was with older versions 3.0 and before, which produced variable outputs, and thus qualities.
![image](https://github.com/user-attachments/assets/61f85f15-997f-46c1-92ee-280bb6a2bb59)

The old PAV system was usually inferior compared to current PAI (Print as image) only
![image](https://github.com/user-attachments/assets/0c79cdb7-3073-431a-be19-260a57498521)


Look at the size outputs for Adobe and MS own print from explorer (both are same page at same resolution) but very drastic different throughputs in terms of size. Adobe know how to rewrite the PDF output for maximum gain. Acrobat can command line print and thus the best.

![image](https://github.com/user-attachments/assets/262e32b1-90f0-4171-ab5e-3da08a36cf26)

**Comment by joaodaffonsojr** [comment]
Thank you guys ! 
Helped me a lot !
